xarray/core/merge.py

```diff
@@ -547,21 +547,24 @@ def dataset_merge_method(dataset, other, overwrite_vars, compat, join):
 
 
 def dataset_update_method(dataset, other):
-    """Guts of the Dataset.update method
+    """Guts of the Dataset.update method.
 
-    This drops a duplicated coordinates from `other` (GH:2068)
+    This drops a duplicated coordinates from `other` if `other` is not an
+    `xarray.Dataset`, e.g., if it's a dict with DataArray values (GH2068,
+    GH2180).
     """
     from .dataset import Dataset
     from .dataarray import DataArray
 
-    other = other.copy()
-    for k, obj in other.items():
-        if isinstance(obj, (Dataset, DataArray)):
-            # drop duplicated coordinates
-            coord_names = [c for c in obj.coords
-                           if c not in obj.dims and c in dataset.coords]
-            if coord_names:
-                other[k] = obj.drop(coord_names)
+    if not isinstance(other, Dataset):
+        other = OrderedDict(other)
```

Comment:
I am slightly wondering this behavior may change after we finally support #884, i.e. we will not include coordinates in `.key()`.